.github/workflows/pylint.yml

```diff
@@ -0,0 +1,44 @@
+name: Linter
+on: pull_request
+jobs:
+  PyLint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run checks
+        run: |
+          URL="https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/files"
+          PR_FILES=$(curl -s -X GET -G $URL | jq -r '.[] | select(.status != "removed") | .filename')
+          for files in $PR_FILES; do
+            extension="${files##*.}"
+            if [[ $extension == 'py' ]]; then
+              changed_files_pylint+=" ${files}"
+            fi
+          done
+
+          if [[ ! -z ${changed_files_pylint} ]]; then
+            sudo apt-get --no-install-recommends install -y build-essential curl python3-dev python3-pip python3-venv
```

Comment:
You should use the `actions/setup-python` action to set up Python. Also, `curl` has to already be installed at this point, or else the `curl` run above would have failed.

Reply:
It seems to me that this will only complicate the reading of the config. Besides, there is no difference. But if it is so necessary, I will do it through ```actions```.

Reply:
There _is_ a difference. With `actions/setup-python`, you can install a specific version of Python, while here you install whatever the default version of Python is for the current OS, which could change at any time and break everything.

Reply:
Reworked.